src/main/src/main/java/org/geoserver/config/GeoServerDataDirectory.java

```diff
@@ -1234,7 +1234,9 @@ else if ( l.getResource() instanceof WMSLayerInfo ) {
             public URL locateResource(String uri) {
                 URL url = super.locateResource(uri);
                 if(url.getProtocol().equalsIgnoreCase("resource")) {
-                    return fileToUrlPreservingCqlTemplates(urlToResource(url).file());
+                    //GEOS-7025: Just get the path; don't try to create the file
+                    return fileToUrlPreservingCqlTemplates(
+                            Paths.toFile(root(), urlToResource(url).path()));
```

Comment:
Could work, although I'd have expected it to test the type of resource, and check if it existsed, returning it in one case, returning null in the other. Could be that returnig null is bad semantic wise though, no strong opinion